# Awesome Bioinformatics Benchmarks
A curated list of bioinformatics bench-marking papers and resources.
The credit for this format goes to Sean Davis for his [awesome-single-cell](https://github.com/seandavi/awesome-single-cell) repository and Ming Tang for his [ChIP-seq-analysis](https://github.com/crazyhottommy/ChIP-seq-analysis) repository. 
If you have a benchmarking study that is not yet included on this list, please make a [Pull Request](https://github.com/j-andrews7/Awesome-Bioinformatics-Benchmarks/pulls) to add it.

## Rules for Included Papers
 - Papers must be objective comparisons of 3 or more tools/methods.
 - Papers should **not** be from authors showing why their tool/method is better than others.
 - Benchmarking data should be publicly available or simulation code/methods should be obviously stated.
 
Additional guidelines/rules may be added as necessary.

# Tool/Method Sections
Additional sections/sub-sections can be added as needed.


## ChIP-seq

### Peak Callers

## RNA-seq

### Differential Gene Expression

## RNA Microarrays

## Variant Callers

### SNP/Indel Callers

### CNV Callers

